### PR TITLE
Add aria-label to SideNav button

### PR DESCRIPTION
**Pull request by holdenhinkle**
## Description
Closes ticket [#28698](https://github.com/department-of-veterans-affairs/va.gov-team/issues/28698)

Adds `aria-label="In this section menu"` to the button in the `SideNav` component.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by mpelzsherman — APPROVED** [review]

File: src/platform/site-wide/side-nav/components/SideNav.js [hunk 1]
```diff
@@ -147,6 +147,7 @@ class SideNav extends Component {
           )}
           onClick={this.toggleUlClass}
           id="sidenav-menu"
+          aria-label="In this section menu"
```
Comment (Rleahy22) [inline src/platform/site-wide/side-nav/components/SideNav.js:150]: Would it make sense to make the `ul#va-sidenav-ul-container` have `role="menu"` and add `aria-controls="va-sidenav-ul-container"`?

**Review by Rleahy22 — APPROVED** [review]

File: src/platform/site-wide/side-nav/components/SideNav.js (hunk 1 above)
Comment (1Copenut) [reply src/platform/site-wide/side-nav/components/SideNav.js:150]: @Rleahy22 I'd prefer not to add a `role="menu"` attribute. Menus require a full suite of keyboard navigation controls including arrow keys: https://www.w3.org/WAI/GL/wiki/Using_ARIA_menus

A menu might also disrupt the parent/child hierarchy of nested lists.

`aria-controls` has very limited use (JAWS is the only SR that supports it) and falls afoul if the ID it references appears and disappears. I'd opt to skip that one too.

**Review by 1Copenut — APPROVED** [review]
LGTM! Please see my comment in response to your question about role and aria-controls.

**Comment by ddzz** [comment]
An aXe test could be added to `src/platform/site-wide/side-nav/tests/components/SideNav.unit.spec.js`.

**Pull request merged**